README.md

```diff
@@ -1,15 +1,53 @@
 # ForemanPuppet
 
-Adds Puppet External node classification funcionality for [Foreman](theforeman.org).
+[![Gem Version](https://img.shields.io/gem/v/foreman_puppet.svg)](https://rubygems.org/gems/foreman_puppet)
+[![Code Climate](https://codeclimate.com/github/theforeman/foreman_puppet/badges/gpa.svg)](https://codeclimate.com/github/theforeman/foreman_puppet)
+[![GPL License](https://img.shields.io/github/license/theforeman/foreman_ansible.svg)](https://github.com/theforeman/foreman_ansible/blob/master/LICENSE)
 
+This plugin adds Puppet External node classification functionality to [Foreman](theforeman.org).
+
+* Website: [theforeman.org](http://theforeman.org)
+* Support: [Foreman support](http://theforeman.org/support.html)
+
+## Features
+
+* Puppet dashboards
+* Puppet
+    * Environments
+    * Classes
+    * Config groups
+    * Smart Class parameters
+* Host / Hostgroup
+    * Classes assignment
+    * Environment assignment
+    * Puppet proxy assignment
+* Puppet ENC (external node classifier)
+* Smartproxy Puppet status/tab
+* Template hostgroup / environment
+
+## Compatibility
+
+The functionality will be extracted from Foreman 3.0. Therefore, this plugin is only required for 3.0 and onwards.  
+You can install it on Foreman 2.5 to prepare for the Foreman update.
+
+|Foreman version|Plugin version|Notes                                     |
+|---------------|--------------|------------------------------------------|
+| >= 3.0        | ~> 1.0       | Required                                 |
+| >= 2.5        | ~> 1.0       | Optional; replaces Core features         |
+| < 2.5         | -            | Not supported (functionality is in Core) |
 ## Installation
 
 See [How_to_Install_a_Plugin](http://projects.theforeman.org/projects/foreman/wiki/How_to_Install_a_Plugin)
 for how to install Foreman plugins
 
 ## Usage
 
-*Usage here*
+The usage is identical to the prior core functionality:
+
+- Import *environments* and *classes* from a Smartproxy
+- Optionally define *Smart Class* parameters
+- Assign *environment* and *classes* to *Hosts / Hostgroups* 
+- Use the *Puppet ENC* on your Puppetserver(s) to query Foreman
 
 ## TODO
```

Comment:
Are there any noteworthy TODOs, we should add here?

Reply:
Imho TODOs belong in issues, not in a README file.

We could ask folks to reports bugs or issues as Github issues, though.